interview-questions.md

```diff
@@ -7,53 +7,54 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. Rails follows an MVC pattern. What does that mean?
 
-  Your answer:
+  Your answer: While I know MVC stands for model-view-controller, I cant recall what that exactly means. I will do some research on that and get back to you.
 
-  Researched answer:
+  Researched answer: Model-view-controller is a framework that provides default structures for a database, a web service, and web pages. 
 
 
 
 2. What is a relational database? Are there other kinds of databases?
 
-  Your answer:
+  Your answer: A relational database is a database that holds data related to other data such as rows and columns. There are other kinds of databases that are non relational but I will have to look up the exact names of those types of databases.
 
-  Researched answer:
+  Researched answer: A relational database stores data in tables with rows and columns. The relationship between the datatables and fields is called a schema. Relational databases are also called SQL databases. A non-relational database does not use rows and columns, it is usually just the data and the key or type of data it is storing. These databases are known as NoSQL.
+  There are four types of NoSQL databases, Document-oriented, Key-Value Stores, Wide-Column Stores, and Graphs Stores.
 
 
 
 3. What is object-relational mapping?
 
-  Your answer:
+  Your answer: Object-relational mapping is being able to access a particular piece of data from a database using commands like SELECT, FROM, ORDER BY etc.
 
-  Researched answer:
+  Researched answer: Object-relational mapping is coverting data between two incompatible languages in object-oriented programming.
 
 
 
 4. What is a gem?
 
-  Your answer:
+  Your answer: What I can recall about gem is that it is a library that holds code to be able to be installed.
 
-  Researched answer:
+  Researched answer: A library of code functionality that makes it easy to share with others, you install gems by using the command $ gem install .
 
 
 
 5. What are primary keys? Why are they important?
 
-  Your answer:
+  Your answer: I believe primary keys are related to databases but I can't recall the exact relation at this moment, I will be happy to look into it and get back to you on that one.
 
-  Researched answer:
+  Researched answer: Primary keys are used to ID each row in a program. We can assign an ID # as a unique identifer to prevent any mix ups in a datatable. 
```

Comment:
Here is another good example of going a bit deeper to help out understanding. What's the primary key used for in a relational database? Why do we need them? What kind of circumstances do they pop up in?